### PR TITLE
Clean up CSharpOperationFactory class

### PR DESCRIPTION
**Pull request by heejaechang**
changes are pure clean up. no functional changes or big refactorings. just moving things to different files or adding lines and etc.

each changes are committed as its own commit per clean up.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 1]
```diff
@@ -913,1175 +913,5 @@ private static IExpressionStatement CreateBoundExpressionStatementOperation(Boun
             Optional<object> constantValue = default(Optional<object>);
             return new LazyExpressionStatement(expression, isInvalid, syntax, type, constantValue);
         }
-
```
Comment (heejaechang) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: this is pure copy and paste of some methods into its own file.

**Comment by heejaechang** [comment]
@dotnet/analyzer-ioperation @dotnet/roslyn-analysis @dotnet/roslyn-compiler can you take a look? changes are purely clean up. each clean up is checked in as separate commits.

**Comment by heejaechang** [comment]
@AlekseyTs @cston @jcouv @dotnet/roslyn-compiler can you take a look?

**Comment by heejaechang** [comment]
can I get one more compiler dev sign off? @dotnet/roslyn-compiler

**Comment by cston** [comment]
LGTM

**Pull request merged**